ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/filebased/FileManagerFactory.java

```diff
@@ -1,64 +1,29 @@
 package ldes.client.treenodesupplier.repository.filebased;
 
 import ldes.client.treenodesupplier.repository.filebased.exception.CreateDirectoryFailedException;
-import ldes.client.treenodesupplier.repository.filebased.exception.DestroyStateFailedException;
 
 import java.io.IOException;
 import java.nio.file.Files;
-import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.util.stream.Stream;
+import java.util.HashMap;
+import java.util.Map;
 
 public class FileManagerFactory {
-	private static FileManagerFactory instance = null;
-	private final FileManager fileManager;
-	public static final String STATE_DIRECTORY = "state";
-	private static boolean stateDeleted = false;
+	public static final String STATE_FOLDER = "ldes-client";
+	private static final Map<String, FileManager> instances = new HashMap<>();
 
-	public FileManagerFactory() {
-		fileManager = new FileManager();
-	}
-
-	public static synchronized FileManagerFactory getInstance() {
-		if (instance == null) {
+	public static synchronized FileManager getInstance(String instanceName) {
+		return instances.computeIfAbsent(instanceName, s -> {
 			try {
-				Files.createDirectories(Paths.get(STATE_DIRECTORY));
+				Files.createDirectories(Paths.get("%s/%s".formatted(STATE_FOLDER, instanceName)));
```

Comment:
Hmm, how does this "/" behave between the %s? I would expect a child directory but windows has a tendency to use "\" for this. Are you sure you don't need a 'separatorsToSystem'?